url/src/lib.rs

```diff
@@ -677,6 +679,46 @@ impl Url {
         self.slice(self.scheme_end..).starts_with("://")
     }
 
+    /// Return the authority of this URL as an ASCII string.
+    ///
+    /// Non-ASCII domains are punycode-encoded per IDNA if this is the host
+    /// of a special URL, or percent encoded for non-special URLs.
+    /// IPv6 addresses are given between `[` and `]` brackets.
```

Comment:
Are you sure that this documentation is actually how it works? Would be good to add a unit test that covers some of these cases, with at least one IDNA-encoded domain, one percent-encoded host and an IPv6 host.

Reply:
Thanks for thr review, @djc.

Yes, I am sure - the documentation is taken from methods that already state this and are implemented in a similar fashion by taking an internal slice. I've added another commit with a few more doc tests to exercise what is being said here and also added a note on ports for special URLs, which are omitted from the authority if they match the well known scheme port.

Reply:
Thanks for adding the tests!

I'd prefer to have this as unit tests not doctests, otherwise the example code gets a bit too bulky for my taste.

Reply:
Agreed, it's a bit too much for docs. Moved those to unit tests leaving just a few examples and also took the opportunity to add unit tests for `has_authority`.

Reply:
Thanks!